jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -1413,6 +1413,20 @@ public String changeSessionId()
         return session.getId();
     }
 
+    protected void onRequestLog()
+    {
+        // Don't allow pulling more parameters
+        _contentParamsExtracted = true;
+
+        // Reset the status code to what was committed
+        MetaData.Response committedResponse = getResponse().getCommittedMetaData();
+        if (committedResponse != null)
+        {
+            getResponse().setStatus(committedResponse.getStatus());
+            // TODO: Reset the response headers to what they were when committed
+        }
```

Comment:
Let's leave all the restoring of state to a different PR, as it is a different issue

Reply:
I'm only restoring response status code, which needs to happen for BadMessage during HttpParser.
There's still a long list of other restoring needed.
See the test cases.

Reply:
The part 2 PR is much more comprehensive, this is the minimum in my view.
Example: We have the entire prevention of Request.getInputStream() and Request.getReader() usage during RequestLog as well in part 2. (which wasn't even in the old PR yet)
This is literally what the first commit in the old PR was, before you asked for the rest (isMutable, restoring response headers, preventing post-commit changes, etc) that belongs in part 2.

Reply:
I'm thinking we can achieve PR2 with a change to the `_errorSentAndIncludes` field that will prevent mutations rather than needing to restore them.
I'll have a play with that, if you can reduce this PR down to the absolute minimum to fix the reading parameters problem-  ie: move the onCompleted() call to before the request log and then set the parameter extracted boolean within the onCompleted call